### DISCUSSION TITLE
Help - Not Receiving Acknowledgment When Publishing to NATS JetStream’s Subject in FastStream

### DISCUSSION THREAD

**Discussion by Atanusaha143** [ag2ai/faststream#1895] [Q&A] [answered]
I am working with FastStream and trying to publish messages to a NATS JetStream, but I’m not seeing any acknowledgment data returned when I attempt to publish. Below is the code snippet I’m using:

```python
class TestPublisher(ABCPublisher):
    def __init__(self, logger: Logger = Depends(Logger)):
        self._logger = logger
        self._broker = NatsBroker("nats://localhost:4222")

    async def publish(self, stream: str, subject: str, message: dict):
        async with self._broker:
            try:
                ack = await self._broker.publish(stream=stream, subject=subject, message=message)
                print(ack)  # This prints 'None'
            except Exception as e:
                self._logger.error(f"Failed to publish message: {e}")
```
I expected ack to contain some acknowledgment details (e.g., a confirmation that the message was successfully stored in the stream). However, the output is None, which makes it unclear if the message was acknowledged or stored.

**Questions:**

1. Is there a way to retrieve acknowledgment details (like sequence ID or confirmation status) after publishing a message to a NATS JetStream subject?
2. Does FastStream provide this capability for NATS JetStream, or do I need additional configuration to enable it?

Any insights or suggestions would be greatly appreciated. Thank you!

**Comment by Lancetnik** [comment]
@Atanusaha143 sorry, now we have a some architecture mistakes and we lost this functional - https://github.com/airtai/faststream/blob/main/faststream/nats/publisher/producer.py#L80

The problem related to `publish(..., rpc=True)` option and we just can't return something from publish method due it could return RPC response later. On **0.6.0** we removed old rpc functional to prior to `request` usage and we can return confirmation frame from publish method, there is no problem

**Reply by Atanusaha143** [reply]
@Lancetnik So, we are getting this functionality on an upcoming version. Right?

**Reply by Lancetnik** [reply, marked as answer]
> @Lancetnik So, we are getting this functionality on an upcoming version. Right?

Yes

**Comment by Lancetnik** [comment]
Should be merged with #1779 - it is already in the branch
